### PR TITLE
Glossary Terms

### PR DESCRIPTION
**Pull request by darrellodonnell**
I am working to get some closure on definitions and terms that should help simplify a few things. The work is not complete, but this PR starts aims to show what needs to be done (and completes a few things).

One item that popped up as a surprise was that `spec-up` was added to `package.json` and `package-lock.json`. I think that is ok, but I was surprised that it wasn't already there. I suppose nobody is doing builds locally.

### CODE REVIEW

File: spec/appendix.md [hunk 1]
```diff
@@ -1,12 +1,58 @@
 ## Appendix
 
+### Trust Over IP Glossary - Controlled Terms 
+
+NOTE: 
+* The following set of identifiers are from the current [ToIP Glossary](https://docs.google.com/document/d/1fZByfuSOwszDRkE7ARQLeElSYmVznoOyJK4sxRvJpyM/edit?usp=sharing) (Google Doc). These definitions need to be moved over to the (under development) ToIP Terms tool (aka TEv2).
```
Comment (swcurran) [inline spec/appendix.md:6]: It’s not clear what definitions “need to be moved” — the ones in this spec or the ones in the ToIP Glossary.  I think you mean that the “ToIP Glossary definitions are being moved to the (under development) ToIP Terms tool (aka TEv2). 

Suggest you clarify.

File: spec/appendix.md [hunk 2]
```diff
@@ -1,12 +1,58 @@
...
+* TODO: not all ToIP definitions are fully linked here. This whole section is more of an exemplar as of the time of writing.
```
Comment (swcurran) [inline spec/appendix.md:7]: Presumably you mean “not all the relevant ToIP definitions…”.  Presumably, the ToIP Glossary is much bigger than this.

File: spec/appendix.md (hunk 1 above)
Comment (darrellodonnell) [reply spec/appendix.md:6]: That's going to be an ongoing effort as we edit this (and other specs).

File: spec/appendix.md [hunk 3]
```diff
@@ -1,12 +1,58 @@
...
+
+[[def: autonomous identifier]] 
+~ Another term for [[ref: self-certifying identifier (SCID)]]. 
+
+[[def: cryptographic keys]]
+~ A key in cryptography is a piece of information, usually a string of numbers or letters that are stored in a file, which, when processed through a cryptographic algorithm, can encode or decode cryptographic data. Symmetric cryptography refers to the practice of the same key being used for both encryption and decryption. Asymmetric cryptography has separate keys for encrypting and decrypting. These keys are known as the public keys and private keys, respectively. Source: Wikipedia.
```
Comment (swcurran) [inline spec/appendix.md:13]: Scary to go to this level in a spec of this type. Do you really think this is needed?  That said, if it is needed and you are putting a Source, I suggest including the link to the source.

Comment (darrellodonnell) [reply spec/appendix.md:13]: It's a good example of a definition/term that can be left out of this spec and point at the ToIP Glossary. Drummond and I are working with Henk to get the tooling figured. For now I'd prefer to leave it in as we don't have a place to point it. 

This is the entry in the (Google Docs) doc that is a temporary place. 

image

File: package.json [hunk 4]
```diff
@@ -55,6 +55,7 @@
     "merge-stream": "2.0.0",
     "pkg-dir": "4.2.0",
     "prismjs": ">=1.24.0",
+    "spec-up": "^0.10.6",
```
Comment (swcurran) [inline package.json:58]: Hmm…seems weird this is needed. I’ve not done it in a while, but have definitely tested locally and it had been working fine.  Doesn’t seem like it was ever removed.  And the GHA publishing is working fine...

Comment (darrellodonnell) [reply package.json:58]: Kevin recently added the spec-up tooling.

File: spec/appendix.md (hunk 1 above)
Comment (darrellodonnell) [reply spec/appendix.md:6]: relates to this Issue: https://github.com/trustoverip/tswg-did-method-webs-specification/issues/116

File: spec/appendix.md (hunk 3 above)
Comment (swcurran) [reply spec/appendix.md:13]: I would suggest for all of these “defined in the ToIP Glossary” terms, you put the link to the ToIP Glossary.  E.g. for this one use [this link](https://docs.google.com/document/d/1fZByfuSOwszDRkE7ARQLeElSYmVznoOyJK4sxRvJpyM/edit#heading=h.53rzpn1yn6q7) and leave out the reference to Source being Wikipedia.

Or if you don’t want to link to exact header, at least the link to the ToIP Glossary, so we know all the things to be fixed later...

Comment (darrellodonnell) [reply spec/appendix.md]: resolved in recent push

File: package.json (hunk 4 above)
Comment (darrellodonnell) [reply package.json]: removed this in https://github.com/trustoverip/tswg-did-method-webs-specification/pull/117/commits/6cd709751e077208ecee9cdbff527438f3c9ff74

File: spec/appendix.md [hunk 5]
```diff
@@ -1,12 +1,58 @@
...
+~ A party that is not directly involved in the trust relationship between a first party and a second party, but provides supporting services to either or both of them.
+
+[[def: verifiable identifier (VID), VID, VIDs]]
+~ An identifier over which the controller can provide cryptographic proof of control.
+See also: decentralized identifier, [ref: self-certifying identifier (SCID)].
+
+[[def: verification]]
+~ An action an agent (of a principal) performs to determine the authenticity of a claim or other digital object using a cryptographic key.
+
+#### Glossary WORK TO BE DONE
```
Comment (2byrds) [inline spec/appendix.md:34]: Is the idea to wait for this work to be completed before we merge this? For instance we generally would only merge a PR that adds terminology if the terminology has [[ref:]] updates that use the [[def:]] terminology being added. And I think the TODO section shouldn't be merged but could be converted into a set of issues that we could process over time. We could decide to add some to the Spec 1.0 milestone or not.

**Review by 2byrds — CHANGES_REQUESTED** [review]

**Comment by darrellodonnell** [comment]
I would prefer to get the work in now, not wait.

Darrell O'Donnell

***@***.***

[image: __tpx__]

File: spec/appendix.md (hunk 5 above)
Comment (darrellodonnell) [reply spec/appendix.md:34]: I would prefer to get the work in now, not wait.

There are enough "things" that need to be cleaned up that we should start immediately

Comment (2byrds) [reply spec/appendix.md:34]: From our meeting today:
@darrellodonnell will break these Todos out as issues and we'll decide which belong in 1.0 and 1.1 spec milestones.

Comment (darrellodonnell) [reply spec/appendix.md:34]: removed the todo items.

File: package-lock.json [hunk 6]
```diff
@@ -38,6 +38,7 @@
         "merge-stream": "2.0.0",
         "pkg-dir": "4.2.0",
         "prismjs": ">=1.24.0",
+        "spec-up": "^0.10.6",
```
Comment (2byrds) [inline package-lock.json:41]: @darrellodonnell this is what i referred to in the slack message

File: package-lock.json [hunk 7]
```diff
@@ -3567,6 +3568,43 @@
       "resolved": "https://registry.npmjs.org/spdx-license-ids/-/spdx-license-ids-3.0.11.tgz",
       "integrity": "sha512-Ctl2BrFiM0X3MANYgj3CkygxhRmr9mi6xhejbdO960nF6EDJApTYpn0BQnDKlnNBULKiCN1n3w9EBkHK8ZWg+g=="
     },
+    "node_modules/spec-up": {
```
Comment (2byrds) [inline package-lock.json:3571]: This as well.

**Review by 2byrds — CHANGES_REQUESTED** [review]
specup related changes should be removed.

File: package-lock.json (hunk 7 above)
Comment (darrellodonnell) [reply package-lock.json:3571]: ok - removed these (unnecessary) changes to package-lock.json.

File: package-lock.json (hunk 6 above)
Comment (darrellodonnell) [reply package-lock.json:41]: removed

File: spec/appendix.md (hunk 1 above)
Comment (2byrds) [reply spec/appendix.md:6]: @darrellodonnell I believe this should be moved out into a separate issue and PR (its basically a TODO)

File: spec/appendix.md (hunk 2 above)
Comment (2byrds) [reply spec/appendix.md:7]: @darrellodonnell I believe this TODO line should be removed. The addition of any other terms can be included in a future PR.

Comment (darrellodonnell) [reply spec/appendix.md:7]: noted - I have made edits to remove the NOTE as well as removed a few definitions that go too far (i.e. really belong in an external glossary). Added issue to GH as well: https://github.com/trustoverip/tswg-did-method-webs-specification/issues/121 

update coming shortly

**Review by 2byrds — APPROVED** [review]
LGTM! Thank you @darrellodonnell !

File: spec/appendix.md [hunk 8]
```diff
@@ -1,12 +1,31 @@
...
+
+[[def: identifier]]
+~ A single attribute—typically a character string—that uniquely identifies an entity within a specific context (which may be a global context). Examples include the name of a party, the URL of an organization, or a serial number for a man-made thing. Supporting definitions: eSSIF-Lab: a character string that is being used for the identification of some entity (yet may refer to 0, 1, or more entities, depending on the context within which it is being used).
+
+[[def: self-certifying identifier (SCID), SCID, SCIDs]]
+~ A subclass of verifiable identifier ([[ref: VID]]) that is [[def: cryptographically verifiable]] without the need to rely on any [[ref: third party]] for [[ref: verification]] because the [[ref: identifier]] is cryptographically bound to the [[ref: cryptographic keys]] from which it was generated. Also known as: [[ref: autonomous identifier]].
```
Comment (swcurran) [inline spec/appendix.md:15]: I think the [[def: crypto…] should be a a [[ref: …], correct?

**Review by swcurran — CHANGES_REQUESTED** [review]
A quick tweak and this looks ready.

File: spec/appendix.md (hunk 8 above)
Comment (darrellodonnell) [reply spec/appendix.md]: good catch. Caused me to remove a few of the refs that I had removed defs for (e.g. third party, verification). change committed.

**Review by swcurran — APPROVED** [review]

**Pull request merged**